### DISCUSSION TITLE
"Feature not available for users using GPG" - why, and what can I (as a user) do about it?

### DISCUSSION THREAD

**Discussion by AndrewSav** [jesseduffield/lazygit#3605] [Q&A] [answered]
When trying to move a hunk from a prior commit to a new commit, I'm getting this:

![image](https://github.com/jesseduffield/lazygit/assets/658865/a358d552-6f8d-446c-9580-cef6438e28c7)

- This is after I was asked and entered the commit message, if anything, is this the best place to give this error, not _before_ that? I mean I spent some time wording it carefully, and now it seems that it's not possible to apply, seems that might be good if we could prevent this?
- Why is feature not available for users using GPG? What other features are not available to such users?
- How to do what I want to do without using GPG? Preferably, not reversing the entire GPG setup, just this one commit, if lazygit cannot support this

**Comment by stefanhaller** [comment, marked as answer]
> This is after I was asked and entered the commit message, if anything, is this the best place to give this error, not before that?

Yes, this would be nice. It should be easy to do, all it takes is to insert this code

```go
if self.config.UsingGpg() {
	return errors.New(self.Tr.DisabledForGPG)
}
```

in `handlePullPatchIntoNewCommit`, before the call to `OpenCommitMessagePanel`; PRs welcome. (We'd be checking this twice then, which is a bit ugly, but improving the user experience in this way is probably worth it, I'd say.)

> Why is feature not available for users using GPG?

Because it's a bit tricky to implement; see [this comment](https://github.com/jesseduffield/lazygit/blob/ef1daf314af3df584249ac479a6efaa4669f4804/pkg/commands/git_commands/rebase.go#L388) in the code.

> How to do what I want to do without using GPG?

If it is possible to configure your GPG setup so that you don't have to enter a password each time, then you can set lazygit's `git.overrideGpg` config to true to make it work. I'm not familiar with GPG at all, so can't give advice here, but I imagine there must be something similar to ssh-agent to make this possible.

**Comment by AndrewSav** [comment]
Thank you, @stefanhaller!

> If it is possible to configure your GPG setup so that you don't have to enter a password each time

With SSH signing you do not need to enter password each time as long as your ssh-agent keeps your password. I'll give `git.overrideGpg` a go. Do you know what else in lazygit check this flag?

**Reply by stefanhaller** [reply]
> Do you know what else in lazygit check this flag?

All the custom patch commands; amending a commit; resetting a commit's author, or adding a co-author; rewording a commit that is not the current head.

**Reply by AndrewSav** [reply]
Thank you. I tried the option it worked for my workflow. Thank you for taking the time for the detailed answer
